omc3/plotting/utils/windows.py

```diff
@@ -0,0 +1,208 @@
+To check if QtPy is installed, either run :meth:`omc3.plotting.plot_window.is_qtpy_installed`
+or try to initialize one of the windows, which will fail with a TypeError, 
+as they want to call QApplication which is set to `None`.
+
+An exception to all of this is  :meth:`omc3.plotting.plot_window.create_pyplot_window_from_fig`,
+which allows to create a `pyplot` handled window from an already existing figure.
+This way, figures can be created without manager (which makes them resource friendlier)
+and can either be added to a QT-Window or, if not installed, opened by `pyplot`.
+"""
+import sys
+from typing import Dict, List, Tuple
+
+import matplotlib
+from matplotlib import pyplot as plt, rcParams
+from matplotlib.figure import Figure
+
+from omc3.utils import logging_tools
+
+LOG = logging_tools.get_logger(__name__)
+
+# --- optional qtpy import block -----------------------------------------------
+
+try:
+    from qtpy.QtWidgets import (
+        QApplication, QMainWindow, QTabWidget, QVBoxLayout, QWidget
+    )
+except ImportError as e:
+    LOG.debug(f"Could not import QtPy: {str(e)}")
+    QMainWindow, QApplication, QVBoxLayout  = None, None, None
+    QWidget, QTabWidget = object, object
+else:
+    from matplotlib.backends.backend_qt5agg import FigureCanvasQTAgg as FigureCanvas
+    from matplotlib.backends.backend_qt5agg import NavigationToolbar2QT as NavigationToolbar
+    try:
+        matplotlib.use('Qt5agg')
+    except ImportError as e:
+        if not "headless" in str(e):
+            raise
+        LOG.debug(
+            "Could not change mpl to use QT, "
+            "due to headless mode (i.e. no display connected)."
+            )
+
+# ------------------------------------------------------------------------------
+
+
+def is_qtpy_installed():
+    """Returns True if QtPy is installed."""
+    return QMainWindow is not None
+
+
+def log_no_qtpy_many_windows():
+    """Logs a warning that QtPy is not installed and many windows will be opened."""
+
+    LOG.warning(
+        "QtPy is not installed. "
+        "Plots will be shown in individual windows. "
+        "Install QtPy for a more organized representation. "
+    )
+
+
+class PlotWidget(QWidget):
+
+    def __init__(self, *figures: Figure, title: str):
+        """A derived QWidget that contains Matplotlib-Figures,
+        stacked vertically, each with its own toolbar.
+
+        Args:
+            *figures (Figure): Figures to be contained in the widget
+            title (str): Name of the widget
+            skip_super (bool): If True the superclass __init__ will be skipped (ONLY FOR TESTS)
+        """
+        QWidget.__init__(self)   # no super(), because will not work with tests
+
+        self.title: str = title
+        self.figures: Tuple[Figure] = figures
+        self._canvas_toolbar_handles: List[Tuple[FigureCanvas, NavigationToolbar]] = []
+
+        self._layout = QVBoxLayout()
+        self.setLayout(self._layout)
+
+        for figure in figures:
```

Comment:
in `self.figures` maybe? Otherwise I'n not sure why we cache then inside the object attributes

Reply:
Fair enough. I see you changed to use `self.figures` in the end, all fine by me here.